tools/analyzer_plugin/test/unit/util/prop_declaration/shared_util.dart

```diff
@@ -0,0 +1,83 @@
+import 'package:analyzer/dart/analysis/results.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/diagnostic/diagnostic.dart';
+import 'package:analyzer/error/error.dart';
+import 'package:collection/collection.dart';
+import 'package:path/path.dart' as p;
+import 'package:test/test.dart';
+
+import '../../../util/over_react_builder.dart';
+import '../../../util/shared_analysis_context.dart';
+import 'shared_test_source.dart';
+
+Future<ResolvedUnitResult> setUpResult(SharedAnalysisContext sharedContext) async {
+  final libraryFilename = sharedContext.nextFilename();
+  final partFilename = p.setExtension(libraryFilename, '.over_react.g.dart');
+
+  final librarySource = sourceTemplate.replaceSingle('{{PART_PATH}}', partFilename);
+  // Generate the part so we don't have to keep updating it whenever the source file changes.
+  // Doing this here seemed easier than attempting to generate it within the test harness package within tests
+  // (and avoid issues with concurrent tests running builds at the same time),
+  // or generating it and checking it in.
+  final partSource = await generateOverReactPart(libraryPath: libraryFilename, librarySource: librarySource);
+
+  final libraryFullPath = sharedContext.createTestFile(librarySource, filename: libraryFilename);
+  final partFullPath = sharedContext.createTestFile(partSource, filename: partFilename);
+
+  Future<ResolvedUnitResult> getResolvedUnit(String path) async {
+    final result = await sharedContext.collection.contextFor(path).currentSession.getResolvedUnit(path);
+    if (result is! ResolvedUnitResult) {
+      throw Exception('Failed to resolve path $path: $result');
+    }
+    if (!result.exists) {
+      throw Exception('Failed to resolve resolve path $path because it did not exist: ${result.uri}');
+    }
+    return result;
+  }
+
+  List<AnalysisError> filterErrors(List<AnalysisError> errors) => errors
+      .where((e) => e.severity != Severity.info && !e.errorCode.name.toLowerCase().startsWith('unused_'))
+      // FIXME(FED-2015) remove once these are properly ignored in generated code
+      .where((e) => e.errorCode.name.toLowerCase() != 'invalid_use_of_visible_for_overriding_member')
```

Comment:
This was cherry-picked from this PR here: https://github.com/Workiva/over_react/pull/873/commits/c859acbcce79d90594cfec8595787536e533e69b to get CI passing after merging the WIP branch, and will get addressed in that PR (FIXME removed in a later commit here https://github.com/Workiva/over_react/pull/873/commits/008939cfcc3b8314ddd6b8e36f29d218ba2e67e8)